packages/ui/nodes/image/src/ImageElement/ImageElement.tsx

```diff
@@ -1,210 +1,94 @@
-    (w: number) => {
-      const path = findNodePath(editor, element);
-      if (!path) return;
-
-      if (w === nodeWidth) {
-        // Focus the node if not resized
-        select(editor, path);
-      } else {
-        setNodes<TImageElement>(editor, { width: w }, { at: path });
-      }
-    },
-    [editor, element, nodeWidth]
-  );
-
-  const onChangeCaption: ChangeEventHandler<HTMLTextAreaElement> = useCallback(
-    (e) => {
-      const path = findNodePath(editor, element);
-      path &&
-        setNodes<TImageElement>(
-          editor,
-          { caption: [{ text: e.target.value }] },
-          { at: path }
-        );
-    },
-    [editor, element]
-  );
-
-  const captionString = useMemo(() => {
-    return getNodeString(nodeCaption[0] as any) || '';
-  }, [nodeCaption]);
-
   return (
-    <div
-      {...attributes}
+    <Image
       css={styles.root.css}
       className={styles.root.className}
       {...rootProps}
-      {...nodeProps}
     >
       <div contentEditable={false}>
         <figure
           css={styles.figure?.css}
           className={`group ${styles.figure?.className}`}
         >
-          <Resizable
+          <ImageResizable
             data-testid="ImageElementResizable"
             // @ts-ignore
             css={styles.resizable?.css}
             className={styles.resizable?.className}
-            size={{ width, height: '100%' }}
-            maxWidth="100%"
-            lockAspectRatio
-            resizeRatio={align === 'center' ? 2 : 1}
-            enable={{
-              left: ['center', 'left'].includes(align),
-              right: ['center', 'right'].includes(align),
-            }}
             handleComponent={{
               left: (
-                <ImageHandle
+                <Box
                   css={[styles.handleLeft?.css]}
                   className={styles.handleLeft?.className}
                 />
               ),
               right: (
-                <ImageHandle
+                <Box
                   css={styles.handleRight?.css}
                   className={styles.handleRight?.className}
                 />
               ),
             }}
-            handleStyles={{
-              left: { left: 0 },
-              right: { right: 0 },
-            }}
-            onResize={(e, direction, ref) => {
-              setWidth(ref.offsetWidth);
-            }}
-            onResizeStop={(e, direction, ref) => setNodeWidth(ref.offsetWidth)}
-            {...resizeProps}
+            ignoreReadOnly={ignoreReadOnly}
+            align={align}
+            {...resizableProps}
           >
-            <img
+            <ImageImg
               data-testid="ImageElementImage"
               css={styles.img?.css}
               className={styles.img?.className}
-              src={url}
-              alt={captionString}
-              draggable={draggable}
               {...nodeProps}
             />
-          </Resizable>
+          </ImageResizable>
```

Comment:
Would it make sense to keep it as `Resizable` if it is being imported from the Image package already?

Effectively Image.Resizable depending how the import/export.

Reply:
This one differs as it's using `useImageResizable`. All the image components use props hooks and the consumer can opt for his own component (or Resizable) + his props or the props hook if needed.

As for the naming pattern, there are 3 options:
1. Image.Root, Image.<sub>: used by radix
2. Image (root), Image.<sub>: used by headlessui
3. Image (root), Image<sub>: used by chakra-ui, reakit

Any preference?